packages/router-plugin/src/router.state.ts

```diff
@@ -85,17 +85,15 @@ export class RouterState {
 
   private _subscription!: Subscription;
 
-  @Selector()
-  static state<T = RouterStateSnapshot>(state: RouterStateModel<T>) {
-    // The `state` is optional if the selector is invoked before the router
-    // state is registered in NGXS.
-    return state?.state;
+  static state<T = RouterStateSnapshot>() {
+    return createSelector([ROUTER_STATE_TOKEN], (state: RouterStateModel<T>) => {
+      // The `state` is optional if the selector is invoked before the router
+      // state is registered in NGXS.
+      return state?.state;
+    });
```

Comment:
@arturovt Isn't this a breaking change because it is updating the public API?

Reply:
Oh, I see that now that this is mentioned in the PR description. I think that there needs to be a note in the changelog about this breaking change.
Or do you think that there is an alternative like a separate `getRouterState<T>()` dynamic selector exposed by the plugin, and we leave the `RouterState.state` as a property.
OR is there the possibility of:
`RouterState<RouterStateSnapshot>.state` where the T on `RouterState` has a default so that `RouterState.state` still works.

Reply:
Hmmm.... I think I have figured out how to make this a non breaking change:
```suggestion
  static get state() {
    return createSelector([ROUTER_STATE_TOKEN], <T = RouterStateSnapshot>(state: RouterStateModel<T>) => {
      // The `state` is optional if the selector is invoked before the router
      // state is registered in NGXS.
      return state?.state;
    });
```
I made a StackBlitz example here:
https://stackblitz.com/edit/vitejs-vite-rrj1efj4?file=src%2Fcounter.ts

Reply:
Doesn't work, unfortunately:

![image](https://github.com/user-attachments/assets/9c0dd271-7dd3-4685-868d-137743627fbe)